src/maggma/stores/gridfs.py

```diff
@@ -170,20 +169,38 @@ def query(
         if isinstance(criteria, dict):
             criteria = self.transform_criteria(criteria)
 
-        for f in self._collection.find(
-            filter=criteria, skip=skip, limit=limit, sort=sort
-        ):
-            data = f.read()
-
-            metadata = f.metadata
-            if metadata.get("compression", "") == "zlib":
-                data = zlib.decompress(data).decode("UTF-8")
+        prop_keys = set()
+        if isinstance(properties, dict):
+            prop_keys = set(properties.keys())
+        elif isinstance(properties, list):
+            prop_keys = set(properties)
 
-            try:
-                data = json.loads(data)
-            except Exception:
-                pass
-            yield data
+        for doc in self._files_store.query(
+            criteria=criteria, sort=sort, limit=limit, skip=skip
+        ):
+            if properties is not None and prop_keys.issubset(set(doc.keys())):
```

Comment:
Do we want this to be on a doc by doc basis or have a store-level property that defines what goes into the metadata?